src/libs/actions/Session/index.js

```diff
@@ -738,4 +847,7 @@ export {
     reauthenticatePusher,
     invalidateCredentials,
     invalidateAuthToken,
+    enableTwoFactorAuth,
+    disableTwoFactorAuth,
+    validateTwoFactorAuth,
```

Comment:
All the functions use the same xxxData, I think is better to create a base function, which you pass just the API name and the `{enable: boolean}` obj

Reply:
Thanks! Ended up with `toggleTwoFactorAuth`:

```js
function toggleTwoFactorAuth(enable) {
    const optimisticData = [ ... ];

    const successData = [ ... ];

    const failureData = [ ... ];

    API.write(enable ? 'EnableTwoFactorAuth' : 'DisableTwoFactorAuth', {}, {optimisticData, successData, failureData});
}
```